File: backend/lib/vis/mail_test.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;


namespace Generated.Validation
{
    public static class Validators
    {
        public static async Task<ValidationResult> ValidateMail_form(Dictionary<string, object?> data)
        {
            var result = new ValidationResult
            {
                Success = false,
                Errors = new List<ValidationError>(),
                Data = data
            };

            var successData = new Dictionary<string, object?>();
            bool flag = true;
            {
                flag = true;bool flag = true;

if (!data.TryGetValue("user_email", out var rawValue) || rawValue == null)
{
    result.Errors.Add(new ValidationError
    {
        Path = "user_email",
        Message = "Invalid value."
    });
    flag = false;
}
else if (rawValue is not string value)
{
    result.Errors.Add(new ValidationError
    {
        Path = "user_email",
        Message = "Invalid value."
    });
    flag = false;
}
else
{
    if (!System.Text.RegularExpressions.Regex.IsMatch(value, "^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9]?(?:\\.[a-zA-Z0-9]?)*$"))
    {
        result.Errors.Add(new ValidationError
        {
            Path = "user_email",
            Message = "Invalid value."
        });
        flag = false;
    }

    if (flag)
    {
        var domain = value.Split('@').Last();
        if (!new List<string>(["gmail.com","outlook.com"]).Contains(domain))
        {
            result.Errors.Add(new ValidationError
            {
                Path = "user_email",
                Message = "Email domain not allowed"
            });
            flag = false;
        }
    }
    value = value.Trim();
    data["user_email"] = value;
}if (flag)
                {
                    successData["user_email"] = data["user_email"];
                }
            }

            if (result.Errors.Count == 0)
            {
                result = new ValidationResult
                {
                    Success = true,
                    Errors = new List<ValidationError>(),
                    Data = successData
                };
            }

            return result;
        }
    }
}

File: frontend/lib/vis/mail_test.js




export const validateMail_form = async (data) => {
    let result = {
        success: false,
        errors: [],
        data: data
    };

    const successData = {};
    let flag = true;
    {
        flag = true;flag = true;

if (data.user_email === undefined || data.user_email === null) {
    result.errors.push({
        path: "user_email",
        message: "Invalid value."
    });
    flag = false;
} else if (typeof data.user_email !== "string") {
    result.errors.push({
        path: "user_email",
        message: "Invalid value."
    });
    flag = false;
} else {
    if (!new RegExp("^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9]?(?:\\.[a-zA-Z0-9]?)*$").test(data.user_email)) {
        result.errors.push({
            path: "user_email",
            message: "Invalid value."
        });
        flag = false;
    }

    if (flag) {
        const domain = data.user_email.split('@').pop();
        const allowedDomains = ["gmail.com","outlook.com"];
        if (!allowedDomains.includes(domain)) {
            result.errors.push({
                path: "user_email",
                message: "Email domain not allowed"
            });
            flag = false;
        }
    }
    data.user_email = data.user_email.trim();
}if (flag) {
            successData.user_email = data.user_email;
        }
    }

    if (result.errors.length === 0) {
        result = {
            success: true,
            errors: [],
            data: successData
        };
    }

    return result;
};